docs/advanced-usage.md

```diff
@@ -131,3 +132,57 @@ staticAnalysis {
 Please note that this is not yet supported for Detekt.
 
 [penaltyextensioncode]: https://github.com/novoda/gradle-static-analysis-plugin/blob/master/plugin/src/main/groovy/com/novoda/staticanalysis/PenaltyExtension.groovy
+
+
+### Consume rules from artefact     
+
+In order to reuse your rules among multiple projects or to easily use an open source rule set, we added support for consuming the 
+rules for all supported tools from a maven artefact. An example of how to create and publish your rules as artefact can be 
+found [here](https://github.com/novoda/novoda/blob/master/scaffolding/build.gradle). Novoda open sourced their rules to [bintray](https://bintray.com/novoda/maven/static-analysis-rules) including
+configurations for Checkstyle, PMD, FindBugs, detekt and Android Lint.
+
+
+To consume the rules from an artefact you need to configure the `rules` closure pointing to a maven artefact and use it when configuring the tools, e.g.
```

Comment:
This is not very clear to me. We need to define:
1 - what a rule artifact is: it's just a bundle of files that is published on a maven repository as artifact.
2 - how you define a dependency from a rule artifact: in order to access the files inside a rule artifact you have to first define an entry in the `rules {}` extension of the plugin. An entry is defined by a name and a maven coordinate, as follows:
```
staticAnalysis {
    rules {
        novoda {
            maven 'com.novoda:static-analysis-rules:0.2'
        }
    }
}
```
3 - once you have defined a rule artifact you can access the files inside it specifying the path of the file inside the bundle, eg:
```
def modules = rules.novoda['checkstyle-modules.xml']
```
Note that `modules` is defined as [`TextResource`](https://docs.gradle.org/current/dsl/org.gradle.api.resources.TextResource.html), that is a read-only body of text backed by a string, file, archive entry, or other source. Some of the tools already accepts `TextResource` as value for some of their configuration, while in other cases you have to transform a `TextResource` into a `File` or a path. Some examples of using a rule artifact in the supported tools is provided below:
- checkstyle sample
- pmd sample
- findbugs sample
- detekt
- lintoptions